src/marketplace/MakeATankSaleView.js

```diff
@@ -87,42 +75,47 @@ class MakeATankSaleView extends React.Component<Props, State> {
 			toast.error("Can't sell last tank");
 			return;
 		}
-		const responsePromise = makeASale(this.state.userId, this.state.salePrice, this.state.tankBeingSoldId, 'tank', 1);
-		responsePromise.then(
-			response => response.json().then(data => {
-				if (response.status !== 201) {
-					console.log(response.status);
-					toast.error(data.msg);
-					console.log(data);
-				}
-				else {
-					toast.success("Tank Listed!");
-					//lets get the new tanks that we have since we lost the current one
+
+		makeASale(
+			this.state.userId, 
+			this.state.salePrice, 
+			this.state.tankBeingSoldId, 
+			"tank", 
+			1,
+			success => {
+				if (success) {
```

Comment:
if we handle errors in the API call, then we can remove all of these `if (success)` checks too

Reply:
Good idea.